Controller/MediaController.php

```diff
@@ -44,27 +45,28 @@ public function getMedia($id)
     /**
      * @throws NotFoundHttpException
      *
-     * @param string $id
-     * @param string $format
+     * @param Request $request
+     * @param string  $id
+     * @param string  $format
      *
      * @return Response
      */
-    public function downloadAction($id, $format = 'reference')
+    public function downloadAction(Request $request, $id, $format = 'reference')
```

Comment:
Do you mean like removing all parameters and get them using `func_get_args()`? The only problem I see with that is the fact that AFAIK Symfony only sends the request if you type hint that parameter.